src/Microsoft.Management.Deployment/PackageCatalogReference.cpp

```diff
@@ -15,42 +15,42 @@
 #include <winget/GroupPolicy.h>
 #include <AppInstallerErrors.h>
 #include <AppInstallerStrings.h>
+#include <winget/UserSettings.h>
 #include <Helpers.h>
 
 namespace winrt::Microsoft::Management::Deployment::implementation
 {
-    namespace
+    void PackageCatalogReference::Initialize(winrt::Microsoft::Management::Deployment::PackageCatalogInfo packageCatalogInfo, ::AppInstaller::Repository::Source sourceReference)
     {
-        std::string GetCallerName()
-        {
-            // See if caller name is set by caller
-            static auto callerName = GetComCallerName("");
+        static constexpr winrt::Windows::Foundation::TimeSpan s_PackageCatalogUpdateIntervalDelay = 168h; //1 week
+
+        m_info = packageCatalogInfo;
+        m_sourceReference = std::move(sourceReference);
+        m_packageCatalogBackgroundUpdateInterval = ::AppInstaller::Settings::User().Get<::AppInstaller::Settings::Setting::AutoUpdateTimeInMinutes>();
 
-            // Get process string
-            if (callerName.empty())
+        bool shouldDelayBackgroundUpdateInterval = false;
+        try
+        {
+            auto [hrGetCallerId, callerProcessId] = GetCallerProcessId();
+            THROW_IF_FAILED(hrGetCallerId);
+            if (callerProcessId != GetCurrentProcessId())
             {
-                try
+                // OutOfProc case, we check for explorer.exe
+                auto callerNameWide = AppInstaller::Utility::ConvertToUTF16(GetCallerName());
+                auto processName = AppInstaller::Utility::ConvertToUTF8(std::filesystem::path{ callerNameWide }.filename().wstring());

```

Comment:
Please move this detection to a shared utility function as I suspect we will need it again (and I want to make it very explicit that we are doing this).

At the same time, wrap it in a `try/catch` so that we can ensure that an unexpected caller value being set doesn't blow up `path` parsing or something.  The values we are targeting should be ok, and so any exception can be treated as not the case we are looking for.